### DISCUSSION TITLE
Splitjoin parents

### DISCUSSION THREAD

**Discussion by IndianBoy42** [Wansmer/treesj#130] [Q&A]
Asking as a discussion in case i've missed this being an existing feature (i have a feeling i misunderstood or dont use recursive preset properly)

Below is the same line of code with two different cursor positions
```lua
M.doplug = function(name, mode, esc) feedkeys(plug_char .. |name, mode, esc) end -- 1
M.doplug = function(name, mode, esc) feed|keys(plug_char .. name, mode, esc) end -- 2
```
Doing `treesj.toggle()` in each of those cases has different behaviour, the first splits the argument list, the second splits the function into 3 lines:
```lua
M.doplug = function(name, mode, esc) feedkeys(
  plug_char .. name,
  mode,
  esc
) end
M.doplug = function(name, mode, esc)
  feedkeys(plug_char .. name, mode, esc)
end
```
Is there a way to get the behaviour of the second case when the cursor is in the position of the first case? The closest i have found is `vif` (using treesitter textobjects) then use `treesj.toggle()`, this is basically equivalent to jumping to the beginning of the function.

In general a way to split/join the parents of deeply nested structures would be nice. 

If it doesn't exist then how could it be implement? I think you could use `vim.v.count1`. Alternatively if an API was given that searched for all the possible splitjoin nodes then we could integrate with flash.nvim to use labels to choose which node to splitjoin.

**Comment by Wansmer** [comment]
> Is there a way to get the behaviour of the second case when the cursor is in the position of the first case? The closest i have found is `vif` (using treesitter textobjects) then use `treesj.toggle()`, this is basically equivalent to jumping to the beginning of the function.

You can specify `enable = false` for `arguments`. In this case, the search will go up the node tree.

```lua
    require('treesj').setup({
      langs = {
        lua = {
          arguments = {
            both = {
              enable = false,
            },
          },
        },
      },
    })
```

> If it doesn't exist then how could it be implement? I think you could use vim.v.count1. Alternatively if an API was given that searched for all the possible splitjoin nodes then we could integrate with flash.nvim to use labels to choose which node to splitjoin.

There are no plans to add functionality for highlighting all nodes. In neovim there are many possibilities to move to the desired location (including treesitter based plugins). I don't think it's right to duplicate such functionality in `treesj`.

**Reply by IndianBoy42** [reply]
I don't really want to disable it for arguments completely id just like to be able to choose which node gets operated on, dynamically.

I actually have a local fork that basically implements what I've asked. Giving `<count>` before the treesj mapping will toggle the `<count>`th parent formattable node, and a separate keybinding exists which calls out to `flash.nvim` for labelling and selecting the node, so the core of the functionality is not duplicated, and is in fact customizable to use any other plugin. I think this is a bit more convenient than jumping to somewhere else, using treesj, and then jumping back

**Reply by Wansmer** [reply]
If you have a working example, you can post a PR or show the code here in the discussions. It will be useful for other members who need such functionality.

**Reply by IndianBoy42** [reply]
Definitely will after a bit of refining

**Reply by IndianBoy42** [reply]
#135
